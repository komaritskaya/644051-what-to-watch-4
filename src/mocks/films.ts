import nanoid from 'nanoid';

const MOVIES_COUNT = 8;

const TITLES = [
  `Avatar`,
  `Aviator`,
  `The Grand Budapest Hotel`,
  `Bohemian Rhapsody`,
  `Dardjeeling Limited`,
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Johnny English`,
  `Macbeth`,
  `Midnight Special`,
  `Mindhunter`,
  `Moonrise Kingdom`,
  `No Country for Old Men`,
  `Orlando`,
  `Player Poster`,
  `Pulp Fiction`,
  `Revenant`,
  `Seven Years in Tibet`,
  `Shutter Island`,
  `Snatch`,
  `The Grand Budapest Hotel`,
  `War of the Worlds`,
  `We Need to Talk About Kevin`,
  `What We Do in the Shadows`,
];

const POSTERS = [
  `avatar.jpg`,
  `aviator.jpg`,
  `bohemian-rhapsody.jpg`,
  `dardjeeling-limited.jpg`,
  `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `johnny-english.jpg`,
  `macbeth.jpg`,
  `midnight-special.jpg`,
  `mindhunter.jpg`,
  `moonrise-kingdom.jpg`,
  `no-country-for-old-men.jpg`,
  `orlando.jpg`,
  `player-poster.jpg`,
  `pulp-fiction.jpg`,
  `revenant.jpg`,
  `seven-years-in-tibet.jpg`,
  `shutter-island.jpg`,
  `snatch.jpg`,
  `the-grand-budapest-hotel-poster.jpg`,
  `war-of-the-worlds.jpg`,
  `we-need-to-talk-about-kevin.jpg`,
  `what-we-do-in-the-shadows.jpg`,
];

const GENRES = [
  `musical`,
  `western`,
  `drama`,
  `comedy`,
  `cartoon`,
  `mystery`,
  `horror`,
];

const getRandomInt = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min + 1)) + min;
};

export const getRandomItemFromArray = (arr) => {
  return arr[getRandomInt(0, arr.length - 1)];
};

const generateMovie = () => ({
  id: nanoid(),
  title: getRandomItemFromArray(TITLES),
  poster: getRandomItemFromArray(POSTERS),
  genre: getRandomItemFromArray(GENRES),
  year: getRandomInt(1980, 2020),
});

const generateMovies = (count) => {
  return new Array(count)
    .fill(``)
    .map(generateMovie);
};

export const movies = generateMovies(MOVIES_COUNT);

export const currentMovie = getRandomItemFromArray(movies);
