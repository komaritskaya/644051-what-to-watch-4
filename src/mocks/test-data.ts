import moment from 'moment';

export const movies = [
  {
    id: `123`,
    title: `Some Film`,
    poster: `some-poster.jpg`,
    genre: `comedy`,
    year: 2000,
    director: `Some Director`,
    cast: [`Actor One`, `Actor Two`],
    cover: `some-bg-poster.jpg`,
    description: `Damn good film`,
    trailer: `some-trailer.mp4`,
    duration: moment.duration(100, `minutes`),
    previewImage: `some-preview-image.jpg`,
    background: `#000`,
    video: `some-video.mp4`,
    isFavorite: true,
    rating: 5.0,
    scoresCount: 50,
  },
  {
    id: `456`,
    title: `Another Film`,
    poster: `another-poster.jpg`,
    genre: `drama`,
    year: 2000,
    director: `Another Director`,
    cast: [`Actor One`, `Actor Two`],
    cover: `another-bg-poster.jpg`,
    description: `Awful film`,
    duration: moment.duration(60, `minutes`),
    trailer: `another-trailer.mp4`,
    previewImage: `another-preview-image.jpg`,
    background: `#fff`,
    video: `another-video.mp4`,
    isFavorite: false,
    rating: 10.0,
    scoresCount: 70,
  },
];

export const tabs = [
  {
    name: `Overview`,
    link: `/`,
  },
  {
    name: `Details`,
    link: `/details`
  },
  {
    name: `Reviews`,
    link: `/reviews`,
  }
];
