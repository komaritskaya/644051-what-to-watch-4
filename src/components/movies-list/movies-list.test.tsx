import React from 'react';
import renderer from 'react-test-renderer';
import MoviesList from './movies-list';
import {movies} from '../../mocks/films';

it(`Should Movies List component render correctly`, () => {
  const tree = renderer
    .create(<MoviesList
      movies={movies}
      onTitleClick={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
