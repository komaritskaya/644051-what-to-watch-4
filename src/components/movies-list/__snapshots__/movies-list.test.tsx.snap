// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Movies List component render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Some Film"
        height="175"
        src="img/some-poster.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Some Film
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Another Film"
        height="175"
        src="img/another-poster.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Another Film
      </a>
    </h3>
  </article>
</div>
`;
